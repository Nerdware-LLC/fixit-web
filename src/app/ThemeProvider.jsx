import React from "react";
import { ThemeProvider as MTP, createTheme } from "@mui/material/styles";
import CssBaseline from "@mui/material/CssBaseline";
import { any } from "../types";

export const ThemeProvider = ({ children }) => {
  const theme = createTheme(THEME);

  return (
    <MTP theme={theme}>
      <CssBaseline />
      {children}
    </MTP>
  );
};

const THEME = {
  description: "dark",
  palette: {
    mode: "dark",
    primary: {
      light: "#f89a35",
      main: "#f78103", // Fixit orange
      dark: "#ac5a02"
    },
    secondary: {
      light: "#5381ff",
      main: "#2962ff", // blue
      dark: "#1c44b2"
    },
    success: {
      light: "#33d373",
      main: "#00c851",
      dark: "#008c38"
    },
    error: {
      main: "rgba(255, 115, 115, 0.87)"
    },
    info: {
      main: "#ff80ab" // AKA NOTIFICATION
    },
    text: {
      primary: "rgba(255, 255, 255, 0.87)",
      disabled: "rgba(255, 255, 255, 0.38)",
      hint: "rgba(255, 255, 255, 0.54)", // AKA PLACEHOLDER
      labels: "#f7a44a" // desaturated Fixit orange (CUSTOM KEY)
    },
    background: {
      paper: "#353535", // AKA SURFACE
      default: "#121212",
      backdrop: "rgba(0, 0, 0, 0.5)" // (CUSTOM KEY)
    }
  }
};

ThemeProvider.propTypes = {
  children: any.isRequired
};
