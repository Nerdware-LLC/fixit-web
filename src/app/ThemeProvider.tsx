import React from "react";
import MuiThemeProvider from "@mui/material/styles/ThemeProvider";
import CssBaseline from "@mui/material/CssBaseline";
import { createTheme, type PaletteOptions } from "@mui/material/styles";
import { themeStore } from ".";
import { any } from "../types";

export const ThemeProvider = ({ children }: { children: React.ReactNode }) => {
  const theme = themeStore.useSubToStore();

  return (
    <MuiThemeProvider theme={THEMES[theme]}>
      <CssBaseline />
      {children}
    </MuiThemeProvider>
  );
};

/**
 * `ThemeProvider` merges this theme object with the @mui/material
 * default theme:
 *
 * https://mui.com/material-ui/customization/default-theme/
 */
const THEMES = Object.fromEntries(
  ["DARK", "LIGHT"].map((themeModeName) => [
    themeModeName,
    createTheme({
      palette: {
        mode: themeModeName.toLowerCase() as PaletteOptions["mode"],
        primary: {
          light: "#f89a35",
          main: "#f78103", // Fixit orange
          dark: "#ac5a02"
        },
        secondary: {
          light: themeModeName === "DARK" ? "#00FFDC" : "#625AFA",
          main: themeModeName === "DARK" ? "#00FFDC" : "#625AFA",
          dark: themeModeName === "DARK" ? "#00FFDC" : "#625AFA"
        },
        success: {
          light: "#33d373",
          main: "#00c851",
          dark: "#008c38"
        },
        error: {
          main: "rgba(255, 115, 115, 0.87)"
        },
        info: {
          main: "#5796D6" // AKA NOTIFICATION
        },
        text: {
          primary: themeModeName === "DARK" ? "rgba(255, 255, 255, 0.87)" : "rgba(0, 0, 0, 0.87)",
          secondary: themeModeName === "DARK" ? "#00FFDC" : "#625AFA",
          disabled: "rgba(255, 255, 255, 0.38)",
          hint: "rgba(255, 255, 255, 0.54)", // AKA PLACEHOLDER
          labels: "#f7a44a" // desaturated Fixit orange (CUSTOM KEY)
        } as any, // <-- Custom theme properties are fine.
        background: {
          paper: themeModeName === "DARK" ? "#353535" : "#cccccc", // AKA SURFACE
          default: themeModeName === "DARK" ? "#121212" : "#f6f6f6",
          backdrop: "rgba(0, 0, 0, 0.5)" // (CUSTOM KEY)
        } as any // <-- Custom theme properties are fine.
      },
      typography: {
        fontFamily: '"Roboto", sans-serif'
      }
    })
  ])
);

ThemeProvider.propTypes = {
  children: any.isRequired
};
