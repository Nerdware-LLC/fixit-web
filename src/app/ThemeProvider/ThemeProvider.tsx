import CssBaseline from "@mui/material/CssBaseline";
import MuiThemeProvider from "@mui/material/styles/ThemeProvider";
import { usePageLayoutContext } from "@app/PageLayoutContext/usePageLayoutContext";
import { themeStore } from "@app/apolloCache/reactiveVars/themeStore";
import { THEMES } from "./themes";
import type { Theme } from "@mui/material/styles";

export const ThemeProvider = ({ children }: ThemeProviderProps) => {
  const themeName = themeStore.useSubToStore();
  const { isMobileUserAgent, isMobilePageLayout } = usePageLayoutContext();

  const theme: Theme = {
    ...THEMES[themeName || "DARK"], // A fallback themeName for added safety (will crash if not set)
    variables: {
      isMobileUserAgent,
      isMobilePageLayout,
    },
  };

  return (
    <MuiThemeProvider theme={theme}>
      <CssBaseline />
      {children}
    </MuiThemeProvider>
  );
};

export type ThemeProviderProps = { children: React.ReactNode };
