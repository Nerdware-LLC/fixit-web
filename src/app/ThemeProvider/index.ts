export { ThemeProvider } from "./ThemeProvider";
export * from "./themes";
