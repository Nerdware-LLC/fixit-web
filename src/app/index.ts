export { App } from "./App";

// Also export apollo-cache reactive vars:
export * from "./apolloCache/reactiveVars";
