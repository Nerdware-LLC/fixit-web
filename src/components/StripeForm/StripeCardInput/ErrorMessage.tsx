import Text from "@mui/material/Typography";
import { styled } from "@mui/material/styles";

export const ErrorMessage = ({
  error,
  children,
  ...props
}: { error?: string | null } & React.ComponentPropsWithoutRef<"div">) => (
  <ErrorBox errorExists={!!error} {...props}>
    <ErrorText>{children ?? error ?? ""}</ErrorText>
  </ErrorBox>
);

// prettier-ignore
const ErrorBox = styled("div")<{ errorExists: boolean }>(({ errorExists }) => `
  visibility: ${errorExists ? "visible" : "hidden"};
  box-sizing: border-box;
  height: 1.5rem;
  padding: 0.25rem 0;
  white-space: nowrap;
`);

// prettier-ignore
const ErrorText = styled(Text)(({ theme }) => `
  color: ${theme.palette.error.main};
  position: relative;
  animation-duration: 1s;
  animation-name: slide-down;
  @keyframes slide-down {
    from {
      top: -1rem;
      opacity: 0;
    }
    to {
      top: 0;
      opacity: 1;
    }
  }
`);
