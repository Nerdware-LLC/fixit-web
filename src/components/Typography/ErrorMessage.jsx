import React from "react";
import styled from "styled-components";
import { Text } from "./Text";
import { node, string } from "../../types";

export const ErrorMessage = ({ error, children, ...props }) => (
  <StyledErrorBox error={error} {...props}>
    <StyledErrorText>{children ?? error ?? ""}</StyledErrorText>
  </StyledErrorBox>
);

const StyledErrorBox = styled.div`
  box-sizing: border-box;
  visibility: ${({ error }) => (error ? "visible" : "hidden")};
  height: 1.5rem;
  padding: 0.25rem 0;
  white-space: nowrap;
`;

const StyledErrorText = styled(Text)`
  color: ${({ theme }) => theme.palette.error.main};
  position: relative;
  animation-duration: 1s;
  animation-name: slide-down;
  @keyframes slide-down {
    from {
      top: -1rem;
      opacity: 0;
    }
    to {
      top: 0;
      opacity: 1;
    }
  }
`;

ErrorMessage.propTypes = {
  error: string,
  children: node
};
