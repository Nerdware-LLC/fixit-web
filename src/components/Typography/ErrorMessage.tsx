import React from "react";
import { useTheme } from "@mui/material/styles";
import styled from "@emotion/styled";
import { Text } from "./Text";
import { node, string } from "../../types";

export const ErrorMessage = ({
  error,
  children,
  ...props
}: { error?: string | null } & React.ComponentPropsWithoutRef<"div">) => {
  const { palette } = useTheme();

  return (
    <StyledErrorBox style={{ visibility: !!error ? "visible" : "hidden" }} {...props}>
      <StyledErrorText style={{ color: palette.error.main }}>
        {children ?? error ?? ""}
      </StyledErrorText>
    </StyledErrorBox>
  );
};

const StyledErrorBox = styled.div`
  box-sizing: border-box;
  height: 1.5rem;
  padding: 0.25rem 0;
  white-space: nowrap;
`;

const StyledErrorText = styled(Text)`
  position: relative;
  animation-duration: 1s;
  animation-name: slide-down;
  @keyframes slide-down {
    from {
      top: -1rem;
      opacity: 0;
    }
    to {
      top: 0;
      opacity: 1;
    }
  }
`;

ErrorMessage.propTypes = {
  error: string,
  children: node
};
