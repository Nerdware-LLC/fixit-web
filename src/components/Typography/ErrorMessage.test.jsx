import renderer from "react-test-renderer";
import { ErrorMessage } from "./ErrorMessage";
import { ThemeProvider } from "../../app/ThemeProvider";

it("renders correctly", () => {
  // ThemeProvider is necessary to be able to access prop.theme in styled css.
  const tree = renderer
    .create(
      <ThemeProvider>
        <ErrorMessage error="This is a TEST error message." />
      </ThemeProvider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
