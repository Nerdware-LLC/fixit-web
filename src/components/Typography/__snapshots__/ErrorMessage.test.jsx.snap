// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-1vh8195"
>
  <p
    className="css-1qyvk2s"
  >
    This is a TEST error message.
  </p>
</div>
`;
